File: Android/NGMobile/app/src/main/java/com/shyam/ngmobile/Model/Member.java
package com.shyam.ngmobile.Model;

import com.shyam.ngmobile.Enums.MemberStatus;

import java.util.Date;

public class Member {
    private String userID;
    private String membershipNo;
    private String fullName;
    private String phoneNumber;
    private String email;
    private String postAddress;
    private String zipCode;
    private String city;
    private String country;
    private String memberType;
    private MemberStatus accountStatus;
    private Date memberExpiryDate;
    private Date gymExpiryDate;
    private boolean firstTimeLogin;

    public Member() {
    }

    public Member(String userID, String membershipNo, String fullName, String phoneNumber,
                  String email, String postAddress, String zipCode, String city, String country,
                  String memberType, MemberStatus accountStatus, Date memberExpiryDate, Date gymExpiryDate, boolean firstTimeLogin) {
        this.userID = userID;
        this.membershipNo = membershipNo;
        this.fullName = fullName;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.postAddress = postAddress;
        this.zipCode = zipCode;
        this.city = city;
        this.country = country;
        this.memberType = memberType;
        this.accountStatus = accountStatus;
        this.memberExpiryDate = memberExpiryDate;
        this.gymExpiryDate = gymExpiryDate;
        this.firstTimeLogin = firstTimeLogin;
    }

    public Member(String userID, String membershipNo, String fullName, String phoneNumber,
                  String email, String memberType, MemberStatus accountStatus,
                  Date memberExpiryDate, Date gymExpiryDate, boolean firstTimeLogin) {
        this.userID = userID;
        this.membershipNo = membershipNo;
        this.fullName = fullName;
        this.phoneNumber = phoneNumber;
        this.email = email;
        this.memberType = memberType;
        this.accountStatus = accountStatus;
        this.memberExpiryDate = memberExpiryDate;
        this.gymExpiryDate = gymExpiryDate;
        this.firstTimeLogin = firstTimeLogin;
    }


    public String getUserID() {
        return userID;
    }

    public String getMembershipNo() {
        return membershipNo;
    }

    public String getFullName() {
        return fullName;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getEmail() {
        return email;
    }

    public String getPostAddress() {
        return postAddress;
    }

    public String getZipCode() {
        return zipCode;
    }

    public String getCity() {
        return city;
    }

    public String getCountry() {
        return country;
    }

    public String getMemberType() {
        return memberType;
    }

    public MemberStatus getAccountStatus() {
        return accountStatus;
    }

    public Date getMemberExpiryDate() {
        return memberExpiryDate;
    }

    public Date getGymExpiryDate() {
        return gymExpiryDate;
    }

    public boolean isFirstTimeLogin() {
        return firstTimeLogin;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public void setPostAddress(String postAddress) {
        this.postAddress = postAddress;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setMemberExpiryDate(Date memberExpiryDate) {
        this.memberExpiryDate = memberExpiryDate;
    }

    public void setGymExpiryDate(Date gymExpiryDate) {
        this.gymExpiryDate = gymExpiryDate;
    }

    public void setFirstTimeLogin(boolean firstTimeLogin) {
        this.firstTimeLogin = firstTimeLogin;
    }


}

File: IOS/NG Mobile App/NG Mobile App/Model/Member.swift
//
//  Member.swift
//  NG Mobile App
//
//  Created by Shyam Vekaria on 7/03/22.
//

import Foundation
import Firebase

struct Member {
    
    init(data: [String:Any]){
        
        self.userID = data["userID"]as? String ?? ""
        self.membershipNo = data["membershipNo"] as? String ?? ""
        self.fullName = data["fullName"] as? String ?? ""
        self.phoneNumber = data["phoneNumber"] as? String ?? ""
        self.email = data["email"] as? String ?? ""
        self.postAddress = data["postAddress"] as? String ?? ""
        self.zipCode = data["zipCode"] as? String ?? ""
        self.city = data["city"] as? String ?? ""
        self.country = data["country"] as? String ?? ""
        self.memberType = data["memberType"] as? String ?? ""
        self.accountStatus = data["accountStatus"] as? String ?? ""
        self.memberExpiryDate = getTime(data: data["memberExpiryDate"] as! Timestamp)
        self.gymExpiryDate = getTime(data: data["gymExpiryDate"] as! Timestamp)
        self.firstTimeLogin = data["firstTimeLogin"] as? Bool ?? true
    }
    
    var userID: String
    var membershipNo: String
    var fullName: String
    var phoneNumber: String
    var email: String
    var postAddress: String
    var zipCode: String
    var city: String
    var country: String
    var memberType: String
    var accountStatus: String
    var memberExpiryDate: Date = Date()
    var gymExpiryDate: Date = Date()
    var firstTimeLogin: Bool = true
    
    var stringMemberExpiryDate: String {
        let formatter = DateFormatter()
        formatter.dateFormat = "dd-MM-yyyy"
        return formatter.string(from: memberExpiryDate)
    }
    
    var stringGymExpiryDate:String {
        let formatter = DateFormatter()
        formatter.dateFormat = "dd-MM-yyyy"
        return formatter.string(from: gymExpiryDate)
    }
    
    private func getTime(data: Timestamp) -> Date{
        let date = data.dateValue()
        
        return Calendar.current.date(byAdding: .hour, value: -3, to: date)!
    }
}
